### PR TITLE
Introduce method: pyenv.versionSupported

### PR DESCRIPTION
**Pull request by nre-ableton**
- Refactor check for pyenv root to method
- Introduce method: pyenv.versionSupported
- Version 0.13.1

### CODE REVIEW

File: src/com/ableton/Pyenv.groovy [hunk 1]
```diff
@@ -61,6 +61,35 @@ class Pyenv implements Serializable {
...
+        script: "${pyenvRoot}/bin/pyenv install --list",
+      )
+
+      // Eliminate whitespace and remove non-version output
+      allVersions.split('\n').each { version ->
+        Object matcher = (version =~ /\s+(\S+)/)
+        if (matcher.find() && matcher.group(1) == pythonVersion) {
+          result = true
+        }
+      }
```
Comment (ala-ableton) [inline src/com/ableton/Pyenv.groovy:87]:
```suggestion
      allVersions.split('\n').each { version ->
        if (version.trim() == pythonVersion) {
          result = true
        }
      }
```
Comparing `"Available versions:"` to `pythonVersion` doesn't hurt.

**Review by ala-ableton — CHANGES_REQUESTED** [review]

File: test/com/ableton/PyenvTest.groovy [hunk 2]
```diff
@@ -46,25 +64,33 @@ class PyenvTest extends BasePipelineTest {
...
+  2.3.7
+'''
+    helper.addShMock('/pyenv/bin/pyenv install --list', mockPyenvVersions, 0)
+    helper.registerAllowedMethod('fileExists', [String]) { return true }
     helper.registerAllowedMethod('isUnix', []) { return true }
 
-    assertThrows(AssertionError) { new Pyenv(script, null).createVirtualEnv('1.2.3') }
+    assertTrue(new Pyenv(script, '/pyenv').versionSupported('2.1.3'))
+    assertFalse(new Pyenv(script, '/pyenv').versionSupported('2.1.3333'))
+    assertFalse(new Pyenv(script, '/pyenv').versionSupported('Available versions:'))
```
Comment (ala-ableton) [inline test/com/ableton/PyenvTest.groovy:87]:
```suggestion
```

**Review by ala-ableton — APPROVED** [review]

**Comment by nre-ableton** [comment]
@ala-ableton Ack, forgot documentation + the singleton stub. ptal, thanks!

File: src/com/ableton/Pyenv.groovy [hunk 3]
```diff
@@ -61,6 +61,39 @@ class Pyenv implements Serializable {
     return venv
   }
 
+  /**
+   * Check if a given Python version is supported by the installed Pyenv version.
```
Comment (ala-ableton) [inline src/com/ableton/Pyenv.groovy:65]:
```suggestion
   * Check if a given Python version is supported by the installed pyenv version.
```

File: vars/pyenv.groovy [hunk 4]
```diff
@@ -12,3 +12,14 @@ import com.ableton.VirtualEnv
 VirtualEnv createVirtualEnv(String pythonVersion, String pyenvRoot = null) {
   return new Pyenv(this, pyenvRoot ?: env.PYENV_ROOT).createVirtualEnv(pythonVersion)
 }
+
+
+/**
+ * Check if a given Python version is supported by the installed Pyenv version.
```
Comment (ala-ableton) [inline vars/pyenv.groovy:18]:
```suggestion
   * Check if a given Python version is supported by the installed pyenv version.
```

File: src/com/ableton/Pyenv.groovy (hunk 3 above)
Comment (nre-ableton) [reply src/com/ableton/Pyenv.groovy]: Fixed in 6020442.

**Review by ala-ableton — APPROVED** [review]

**Comment by nre-ableton** [comment]
@ablbot autosquash

**Pull request merged**